File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int n = needle.size();
        for(int i=0; i<haystack.size();i++) {
            int j = 0;
            for(j=0; j<n; j++) {
                if(haystack[i+j] != needle[j])
                    break;
            }
            if(j==n)
                return i;
        }
        return -1;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    n := len(needle)
    for i:=0; i<len(haystack)+1-n; i++ {
        j := 0
        for j=0; j<n; j++ {
            if haystack[i+j] != needle[j] {
                break
            }
        }
        if j==n {
            return i
        }
    }
    return -1
}

